service/history/replication/executable_history_task.go

```diff
@@ -177,7 +194,7 @@ func (e *ExecutableHistoryTask) MarkPoisonPill() error {
 		return err
 	}
 
-	events, err := serialization.NewSerializer().DeserializeEvents(e.req.Events)
+	events, err := e.EventSerializer.DeserializeEvents(e.eventsBlob)
```

Comment:
e.getDeserializedEvents()?

Reply:
The original code path here only cares about the `events`: even if `newRunEvents` failed to deserialize, the task can still be put into DLQ. Discussed with @wxing1292 offline, we can keep this semantic for now. So I decided to leave the logic as is.

Reply:
My understanding is this will only read from in-memory. If the new run events failed to deserialize, it will be failed to executed and end up in DLQ. But it is up to you here.